packages/react-reconciler/src/ReactFiberCompleteWork.new.js

```diff
@@ -966,6 +972,53 @@ function completeWork(
       bubbleProperties(workInProgress);
       return null;
     }
+    case Profiler: {
+      const didBailout = bubbleProperties(workInProgress);
+      if (!didBailout) {
+        // Use subtreeFlags to determine which commit callbacks should fire.
+        // TODO: Move this logic to the commit phase, since we already check if
+        // a fiber's subtree contains effects. Refactor the commit phase's
+        // depth-first traversal so that we can put work tag-specific logic
+        // before or after committing a subtree's effects.
+        const OnRenderFlag = Update;
+        const OnCommitFlag = Callback;
+        const OnPostCommitFlag = Passive;
+        const subtreeFlags = workInProgress.subtreeFlags;
+        const flags = workInProgress.flags;
+        let newFlags = flags;
+
+        // Call onRender any time this fiber or its subtree are worked on, even
+        // if there are no effects
+        newFlags |= OnRenderFlag;
```

Comment:
Wasn't sure what the desired behavior was here, but I think conceptually this should fire any time one of the duration fields changes?

So it should at least fire for the initial mount.

Maybe for updates, we can check if the subtree flags includes `PerformedWork`?

Reply:
> but I think conceptually this should fire any time one of the duration fields changes?

Not sure if I'm reading too literally, but we'll still want to call `onRender` even if duration happened to be the same– if we did any rendering work in the subtree.

Reply:
#19885